packages/teleport/src/components/SelectFilter/SelectFilter.tsx

```diff
@@ -0,0 +1,334 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+import styled from 'styled-components';
+import { components } from 'react-select';
+import { Flex, Text, ButtonBorder, ButtonIcon, Box } from 'design';
+import { Close, Add } from 'design/Icon';
+import Select, { Option } from 'shared/components/Select';
+
+export default function SelectFilter({
+  applyFilters,
+  appliedFilters,
+  filters,
+  mb = 3,
+}: Props) {
+  const selectWrapperRef = useRef(null);
+  const [showSelector, setShowSelector] = useState(false);
+  const [selectedFilters, setSelectedFilters] = useState<Option[]>(
+    appliedFilters
+  );
+
+  function clearFilters() {
+    setSelectedFilters([]);
+  }
+
+  function deleteLabel(label: string) {
+    const updatedFilters = appliedFilters.filter(o => o.label !== label);
+    applyFilters(updatedFilters);
+  }
+
+  function onFilterApply() {
+    setShowSelector(false);
+    applyFilters(selectedFilters);
+  }
+
+  function handleKeyDown(e: KeyboardEvent) {
+    if (e.key === 'Escape') {
+      setShowSelector(false);
+    }
+  }
+
+  useEffect(() => {
+    setSelectedFilters(appliedFilters);
+  }, [appliedFilters]);
+
+  useEffect(() => {
+    function handleOnClick(e) {
+      // Ignore event for clicking near buttons.
+      if (e.target.closest('button')) {
+        return;
+      }
+
+      // If event is not from inside the select wrapper, close the selector.
+      if (!selectWrapperRef.current?.contains(e.target)) {
+        setShowSelector(false);
+      }
+    }
+
+    window.addEventListener('click', handleOnClick);
+    return () => window.removeEventListener('click', handleOnClick);
+  }, []);
+
+  const $labels = appliedFilters.map((o, key) => (
+    <Label key={key} name={o.label} onClick={() => deleteLabel(o.label)} />
+  ));
+
+  return (
+    <Box mb={mb}>
+      <Flex flexWrap="wrap">
+        <Box style={{ position: 'relative' }}>
+          <AddButton
+            pl={2}
+            pr={3}
+            onClick={() => setShowSelector(!showSelector)}
+            mt={0}
+            mr={3}
+            mb={2}
+          >
+            <Add fontSize={4} mr={1} color="rgba(255,255,255,0.24)" />
+            Add Filters
+          </AddButton>
+          {showSelector && (
+            <Box
+              mt={-2}
+              bg="#fff"
+              borderRadius={2}
+              borderTopLeftRadius={0}
+              style={{ position: 'absolute', zIndex: 1 }}
+            >
+              <Flex
+                justifyContent="space-between"
+                alignItems="center"
+                px={3}
+                pb={0}
+              ></Flex>
```

Comment:
do we need this Flex? Seems as empty div :) hacking our way through css?

Reply:
oops, leftover, this element does nothing :sweat_smile:, removed